crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -715,6 +717,16 @@ pub fn comparator_operand_type_rules(
             }
             Err(format!("Unsupported types for comparison: {} and {}", name_a, name_b))
         }
+        (String(x_size), String(y_size)) => {
+            x_size.unify(y_size, op.location.span, errors, || {
+                TypeCheckError::Unstructured {
+                    msg: format!("Can only compare arrays of the same length. Here LHS is of length {}, and RHS is {} ", x_size, y_size),
```

Comment:
The error message here should mention strings instead of arrays

Reply:
Done